Sources/NIO/EventLoop.swift

```diff
@@ -291,6 +295,74 @@ extension TimeAmount: Comparable {
     }
 }
 
+extension TimeAmount {
+    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds + rhs.nanoseconds)
+    }
+
+    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds - rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount.Value, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs * rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount, rhs: TimeAmount.Value) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds * rhs)
+    }
+}
+
+/// Represents a point in time.
+///
+/// - note: `Time` should not be used to represent an interval
+public struct Time {
```

Comment:
This struct is missing some conformances. I think we need:

- `Equatable` and `Hashable` (the synthesised conformances will be fine)
- `CustomStringConvertible`